src/rp2_common/wifi_settings_connect/doc/SETTINGS_FILE.md

```diff
@@ -0,0 +1,160 @@
+```
+and the equivalent for Pico 2 W (RP2350):
+```
+    picotool load -o 0x103fc000 mywifisettings.bin
+```
+
+## Location of the wifi-settings file
+
+The default location of the file (16kb before the final address in Flash)
+has been chosen because the final three 4kb Flash sectors are already assigned
+a function by the Pico SDK. The Bluetooth library uses two 4kb sectors for storage of
+devices that have been paired by Bluetooth. The final 4kb sector is used for a workaround
+for the RP2350-E10 bug - this sector may be erased when copying a UF2 file to a Pico 2
+via drag-and-drop. Therefore, these three sectors are avoided.
+
+If you wish to store the wifi-settings file at a specific address you can
+do so by setting `-DWIFI_SETTINGS_FILE_ADDRESS=0x....` when running `cmake`.
+The value `0x...` should be an address relative to the start of Flash, so Flash address
+`0x1fc000` corresponds to absolute address `0x101fc000`.
+
+## Backing up a WiFi settings file
+
+picotool can be used to download WiFi settings files from a Pico W:
+```
+    picotool save -r 0x101fc000 0x101fd000 backup.bin
+```
+and Pico 2 W (RP2350):
+```
+    picotool save -r 0x103fc000 0x103fd000 backup.bin
+```
+Characters after the end of the file will be copied (usually either 0x00 or 0xff).
+These can be safely deleted using your text editor. The backup is restored by
+using `picotool load` as described in "Copying the WiFi settings file by USB".
+
+These examples use the default location for the wifi-settings file. If you
+are using a custom location, e.g. building with
+`-DWIFI_SETTINGS_FILE_ADDRESS=0x...`, then
+you would need to substitute the actual address.
+
+# File format details
+
+The file format is very simple so that it can be read by a simple algorithm
+that doesn't require much code space. The parser ignores any line that it
+doesn't understand, and skips any keys that are not known. Here are the rules:
+
+ - The key and the value should be separated only by an `=` character, e.g. `ssid1=HomeWiFi`.
+ - Lines that don't match the form `key=value` are completely ignored;
+   you can add text, comments etc. in order to help you manage your configuration.
+ - On a line that does match `key=value`, whitespace is NOT ignored.
+   Be careful to avoid adding extra spaces around `=`.
+   A space before `=` will be part of the key, and a space after `=` will be part of the value.
+ - Unix and Windows line endings are supported.
+ - The maximum size of the file is 4096 bytes.
+ - Values can contain any printable UTF-8 character.
+ - Keys can also contain any printable UTF-8 character except for '='.
+ - There is no maximum size for a key or a value (except for the file size).
+ - Values can be zero length.
+ - Keys must be at least 1 byte.
+ - If a key appears more than once in the file, the first value is used.
+ - The end of the file is the first byte with value 0x00, 0xff or 0x1a, or the 4097th byte,
+   whichever comes first.
```

Comment:
It's easy to imagine this "settings-file reader" code being useful for multiple projects beyond your wifi-settings-connect library. I wonder if it might be worth splitting it out into a library of its own? (If you have the energy & time, of course :slightly_smiling_face: )

Reply:
> Is there a plan to officially adopt a filesystem as part of the SDK?

See https://github.com/raspberrypi/pico-sdk/labels/filesystem